tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -12426,11 +12426,11 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC, Expr *BitWidth,
          "otherwise this is called without checking language first");
 
   // NOTE: some tests may declare templates.
-  if (DC->isDependentContext()) return true;
+  if (DC->isDependentContext() && !getLangOpts().EnableTemplates) return true;
```

Comment:
I think in this case we should return `false` not `true`. I know that's not how the code originally was, but we should be surfacing an error if template dependent contexts are generated without templates being enabled.